Add a doc page about handling flaky tests (#20691)
opf/openproject: docs/development/testing/continuous-testing-workflow/README.md

Some tests can fail on GitHub actions CI, and pass locally which makes them harder to reproduce, diagnose, and fix.

> These are flaky tests, and they should be fixed as soon as possible as they undermine the trust we have into test results. [Follow our advices to handle flaky tests](../handling-flaky-tests/) for more information.

Reviewer: ```suggestion
These are flaky tests and should be fixed as soon as possible, as they undermine the trust we have in our test results. [Follow our advice on handling flaky tests](../handling-flaky-tests/) for more information.
```
